presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -862,6 +860,14 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Expression
 
             // Sort non-equality predicates by those that can be pushed down and those that cannot
             for (Expression conjunct : EqualityInference.nonInferrableConjuncts(inheritedPredicate)) {
+                if (node.getGroupIdSymbol().isPresent() && SymbolsExtractor.extractUnique(conjunct).contains(node.getGroupIdSymbol().get())) {
+                    // aggregation operator synthesizes outputs for group ids corresponding to the global grouping set (i.e., ()), so we
+                    // need to preserve any predicates that evaluate the group id to run after the aggregation
+                    // TODO: we should be able to infer if conditions on grouping() correspond to global grouping sets to determine whether
+                    // we need to do this for each specific case
+                    postAggregationConjuncts.add(conjunct);
```

Comment:
shouldn't be this block finished with `continue`, as this conjunct could be added in the code below to `pushdownConjucts` anyway?

Reply:
Yes, good catch.